peer/peerstorage.go

```diff
@@ -0,0 +1,103 @@
+package peer
+
+import (
+	"errors"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/lightningnetwork/lnd/kvdb"
+)
+
+var (
+	// peerStorage is the kvdb key where we store other peer's backup data
+	// which they send us.
+	peerStorage = []byte("peer-storage")
+
+	// ErrUninitializedDB is called when the DB is not initialized yet
+	// but a caller calls the `Retrieve` function.
+	ErrUninitializedDB = errors.New("uninitialized Kvdb peer data " +
+		"store")
+)
+
+// KvdbPeerStorage is the kvdb implementation of the PeerDataStore interface.
+type KvdbPeerStorage struct {
```

Comment:
suggest: `PeerStorageDB`

Reply:
I'd leave as is. Having the struct indicate how it implements the interface is a good thing imo.